src/cmds/restic/find.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+
+	"restic"
+	"restic/repository"
+)
+
+// FindFilteredSnapshots yields Snapshots, either given explicitly by `args` or filtered from the list of all snapshots.
+func FindFilteredSnapshots(ctx context.Context, repo *repository.Repository, host string, tags []string, paths []string, args []string) <-chan *restic.Snapshot {
+	out := make(chan *restic.Snapshot)
+	go func() {
+		defer close(out)
+		if len(args) != 0 {
+			var (
+				id         restic.ID
+				usedFilter bool
+				err        error
+			)
+			ids := make(restic.IDs, 0, len(args))
+			// Process all snapshot IDs given as arguments.
+			for _, s := range args {
+				if s == "latest" {
+					id, err = restic.FindLatestSnapshot(repo, paths, tags, host)
+					if err != nil {
+						Warnf("Ignoring %q, no snapshot matched given filter (Paths:%v Tags:%v Host:%v)\n", s, paths, tags, host)
+						usedFilter = true
+						continue
+					}
+				} else {
+					id, err = restic.ParseID(s)
```

Comment:
The code here now requires passing a complete (non abbreviated) snapshot ID, that's not a good user experience:
```
$ bin/restic snapshots       
ID        Date                 Host        Tags        Directory
----------------------------------------------------------------------
4b122706  2017-03-07 11:11:52  mopped                  /home/fd0/shared/work/restic/restic

$ bin/restic tag --add foo 4b122706
Create exclusive lock for repository
Ignoring "4b122706", it is not a snapshot id
No snapshots were modified

$ bin/restic list snapshots
4b12270680254642a8fe140dba8e7ede140e40af855899768133f4e01c0c875e

$ bin/restic tag --add foo 4b12270680254642a8fe140dba8e7ede140e40af855899768133f4e01c0c875e
Create exclusive lock for repository
Modified tags on 1 snapshots
```

There's `restic.FindSnapshot()` which finds the snapshot ID for from a prefix, it was removed in 63db3045c2bb8661a41ebd357984df6794e62d0d.